src/server/services/pipeline-in-progress/validateReadings.js

```diff
@@ -12,13 +12,17 @@ const { log } = require('../../log');
  * @param {Reading[]} arrayToValidate
  * @param {dict} conditionSet used to validate readings (minVal, maxVal, minDate, maxDate, threshold, maxError)
  * @param {string} meterIdentifier identifier of meter being checked
+ * @returns {object} { validReadings, invalidReadings, errMsg }
  */
 function validateReadings(arrayToValidate, conditionSet, meterIdentifier = undefined) {
 	/* tslint:disable:no-string-literal */
 	const { validDates, errMsg: errMsgDate } = checkDate(arrayToValidate, conditionSet['minDate'], conditionSet['maxDate'], conditionSet['maxError'] / 2, meterIdentifier);
 	const { validValues, errMsg: errMsgValue } = checkValue(arrayToValidate, conditionSet['minVal'], conditionSet['maxVal'], conditionSet['maxError'] / 2, meterIdentifier);
 	/* tslint:enable:no-string-literal */
+	const validReadings = validDates.filter(reading => validValues.includes(reading));
+	const invalidReadings = [...invalidDates, ...invalidValues];
```

Comment:
See above for why invalidDates does not seem set. Also, since these are arrays would using concat be more typical?